### PR TITLE
[SPARK-44640][PYTHON][FOLLOW-UP] Update UDTF error messages to include method name

### PR DESCRIPTION
**Pull request by allisonwang-db**
### What changes were proposed in this pull request?

This PR is a follow-up for SPARK-44640 to make the error message of a few UDTF errors more informative by including the method name in the error message (`eval` or `terminate`).

### Why are the changes needed?

To improve error messages.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing tests.

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by allisonwang-db** [comment]
cc @ueshin

File: python/pyspark/worker.py [hunk 1]
```diff
@@ -770,6 +770,7 @@ def verify_result(result):
                         message_parameters={
                             "type_name": type(result).__name__,
                             "value": str(result),
+                            "func": f.__name__,
```
Comment (ueshin) [inline python/pyspark/worker.py:773]: I guess we should do the same thing for `UDTF_RETURN_SCHEMA_MISMATCH` or other error classes if possible?

Comment (allisonwang-db) [reply python/pyspark/worker.py:773]: Good point! I will update the other places.

**Review by ueshin — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by zhengruifeng** [comment]
@allisonwang-db the failure seems related

**Comment by allisonwang-db** [comment]
We need this PR in branch-3.5 as well.

**Comment by ueshin** [comment]
Thanks! merging to master/3.5.

**Comment by ueshin** [comment]
@allisonwang-db There is a conflict with 3.5. Could you help fix it and submit another PR? Thanks.

**Pull request closed without merging**